internal/codespaces/api/api.go

```diff
@@ -186,16 +187,42 @@ type CodespaceEnvironmentConnection struct {
 	HostPublicKeys []string `json:"hostPublicKeys"`
 }
 
+// codespacesListResponse is the request body for the `/user/codespaces` endpoint
+type getCodespacesListResponse struct {
+	Codespaces []*Codespace `json:"codespaces"`
+	TotalCount int          `json:"total_count"`
+}
+
 // ListCodespaces returns a list of codespaces for the user.
-func (a *API) ListCodespaces(ctx context.Context) ([]*Codespace, error) {
+func (a *API) ListCodespaces(ctx context.Context) (codespaces []*Codespace, err error) {
+	for page := 1; ; page++ {
+		response, err := a.fetchCodespaces(ctx, page)
+		if err != nil {
+			return nil, fmt.Errorf("%w", err)
+		}
+		if len(codespaces) >= response.TotalCount || len(response.Codespaces) == 0 {
+			break
+		}
+		codespaces = append(codespaces, response.Codespaces...)
```

Comment:
I know I recommended making break the "special" case. But does this need to come before the break condition now?
```
codespaces = append(codespaces, response.Codespaces...)

if ... {
  break
}
```